### ISSUE TITLE
Blank page - Not found: /

### ISSUE THREAD

**Issue by X3msnake** [pangodream/ESP2SOTA#3]
I get a blank page with Not found:/ after i upload the AP sketch and connect to it.

Any idea why?

**Comment by SlothDpal** [comment]
add handler for root (/) webserver directory:
  server.on("/", HTTP_GET, []() {
    server.sendHeader("Connection", "close");
    server.send(200, "text/plain", "Hello there!");
  });

**Comment by electromechpro** [comment]
Hi,
I have the same problem.  I added the handler SlothDpal suggested and got the "Hello there!" message when connecting to AP ESP2SOTA and pointing a browser to 10.10.10.1.  
There seems to be a missing step(s) in the implementation of this library.  After doing a USB download of the APMode.ino to the ESP32, what does one then do to perform an OTA download?  There are no ports or IP addresses available as with regular Arduino OTA.  I tried connecting to the ESP2SOTA access point and pointing a browser to 10.10.10.1 prior to attempting download, but, unsurprisingly, there is no destination to download to.
I am using Arduino ver. 2.2.1.

**Comment by SlothDpal** [comment]
**http://10.10.10.1** it's a root page of your esp HTTP-server and his handling by this code https://github.com/pangodream/ESP2SOTA/issues/3#issuecomment-1644483740
If you want to update page, you must enter url: **http://10.10.10.1/update**
If you want automating redirect to **http://10.10.10.1/update** u can add meta tag "redirect" to your root www page i.e:
```
//add handler for root (/) webserver directory:
server.on("/", HTTP_GET, {
server.sendHeader("Connection", "close");
    request->send(200, "text/html", "<html><head><meta http-equiv=\"refresh\" content=\"1;URL=/update\" /></head><body></body></html>");
});
```
or
```
//add handler for root (/) webserver directory:
server.on("/", HTTP_GET, {
server.sendHeader("Connection", "close");
    request->send(200, "text/html", "<head><meta http-equiv=\"refresh\" content=\"1;URL=/update\" /></head>");
});
```

**Comment by electromechpro** [comment]
Thank you for your response.  It provided a key step in solving my problem.  I was able to research some more on this and will try to provide for others a summary of what worked for me.
   Here is how I got AP OTA to work using Arduino 2.x and ESP2SOTA library:

1.  I used the example AP_Mode from the ESP2SOTA library I installed (assuming you installed the library), and chose the board and port (USB COM port for first download).  Note the default SSID and password for the access point created in the code.
2. Download the sketch via USB to the ESP32 board.  Upon completion, check serial monitor (115.2K) for access point IP address 10.10.10.1
3. I used a different computer (my desktop did not have Wi-Fi) and connected Wi-Fi to the ESP2SOTA access point.
4. On that different computer I opened the same example from the ESP2SOTA AP_MODE Arduino library, selecting the same board.
5. I added in some code to blink an LED (set a pin as an output in setup, and made it blink, along with a serial print message)
6. ************Now for the tricky parts- secret knowledge reserved for the elite:***************
7. In the Arduino IDE, instead of compiling/downloading the code (nowhere to send it to, right?), select "Sketch" from the top menu items, then "Export Compiled Binary".  I just went with the default prompts.  This action created a "build" subdirectory (folder) in the Arduino sketch directory.  There was a subdirectory for the board (containing necessary board definitions, I believe), containing several files ending in .bin, .elf, and .map.  Two of the .bin files had the words "bootloader" and "partitions" in them, so I ignored them.
8. Then, open a web browser on that computer and enter 10.10.10.1/update in the address bar.  That is where the illustration from the top of the github is stored.  If you get this far, click "Choose File", and navigate to the subdirectory covered in step 7.  Select the .bin file that does not contain the words "bootloader" or "partitions", then select "Update".  My green progress bar almost instantaneously went from 0 to 100%, and shortly thereafter, my LED started blinking and my local serial console started printing my custom message.

   In summary, the sketch can be downloaded OTA with the ESP32 in AP mode.  As the Arduino software does not know where or how to send the data, you must compile and save the file as a .bin file.  This .bin file is then accessed by a web page stored on the ESP, which calls for the download from the Wi-Fi connected computer.

   Please feel free to correct any syntax or conceptual errors herein.  I only wanted to give a summary of how I was able to make this work by filling in the critical steps left out by those to whom this is second nature.
   This is a super useful tool for updating ESP based projects OTA when no Wi-Fi networks are practical.  I am hoping it will get easier for me as I work more with it.
   Thank you again for your help.
